pkg/controllers/state/cluster.go

```diff
@@ -183,6 +187,40 @@ func (c *Cluster) IsNodeNominated(providerID string) bool {
 	return false
 }
 
+// IsNodeUnderutilized returns true if the last pod event on the node was at least consolidateAfter
+// in the past. Also returns a reason to inform the user on how this was calculated.
+func (c *Cluster) IsNodeUnderutilized(providerID string, consolidateAfter time.Duration) (bool, time.Duration, string) {
+	c.mu.RLock()
+	defer c.mu.RUnlock()
+
+	reason := ""
+	lastPodEventTime, ok := c.lastPodEvents[providerID]
+	if !ok {
+		// If the node doesn't have a pod event, we assume it's because either:
+		// 1. Karpenter has restarted
+		// 2. No pods have ever scheduled to the node
+		//
+		// So treat the last pod time as the more recent of
+		// 1. When Karpenter restarted
+		// 2. When the Node was created
+		if c.clock.Since(c.initTime) < c.clock.Since(c.nodes[providerID].NodeClaim.CreationTimestamp.Time) {
```

Comment:
Rather than capturing an init time, does it make more sense to just hydrate `time.Now()` when the cluster state node creation events are first being hydrated and then you can just know if the provider ID doesn't exist in the map that the node hasn't been seen yet by the cluster state informer and so you throw an error or requeue?